src/test/java/uk/gov/ons/fwmt/job_service/rest/impl/UserResourceServiceImplTest.java

```diff
@@ -1,46 +1,188 @@
+      //Given
+      String testAuthNo = "1111";
+      UserDto expectedUserDto = new UserDto();
+
+      when(restTemplate.exchange(any(),any(),any(),eq(UserDto.class),eq(testAuthNo))).thenReturn(responseEntity);
+      when(responseEntity.getStatusCode()).thenReturn(HttpStatus.OK);
+      when(responseEntity.getBody()).thenReturn(expectedUserDto);
+
+      //When
+      Optional<UserDto> result = userResourceService.findByAuthNo(testAuthNo);
+
+      //Then
+      assertTrue(result.isPresent());
+      verify(restTemplate).exchange(any(),any(),any(),eq(UserDto.class),eq(testAuthNo));
     }
 
-    @Test
-    public void findByAlternateAuthNo() {
-        assertEquals("1111", userResourceService.findByAlternateAuthNo("2222").get().getAuthNo());
+  @Test
+  public void authNoNotFound() {
+    //Given
+    String testAuthNo = "1111";
+    when(restTemplate.exchange(any(),any(),any(),eq(UserDto.class),eq(testAuthNo))).thenThrow(new HttpClientErrorException(HttpStatus.BAD_REQUEST));
+
+    //When
+    Optional<UserDto> result = userResourceService.findByAuthNo(testAuthNo);
+
+    //Then
+    assertFalse(result.isPresent());
+    verify(restTemplate).exchange(any(),any(),any(),eq(UserDto.class),eq(testAuthNo));
+  }
+
+  @Test
+  public void findByAlternateAuthNo() {
+    //Given
+    String testAltAuth = "2222";
+    UserDto expectedUserDto = new UserDto();
+    when(restTemplate.exchange(any(),any(),any(),eq(UserDto.class),eq(testAltAuth))).thenReturn(responseEntity);
+    when(responseEntity.getStatusCode()).thenReturn(HttpStatus.OK);
+    when(responseEntity.getBody()).thenReturn(expectedUserDto);
+
+    //When
+    Optional<UserDto> result = userResourceService.findByAlternateAuthNo(testAltAuth);
+
+    //Then
+    assertTrue(result.isPresent());
+    verify(restTemplate).exchange(any(),any(),any(),eq(UserDto.class),eq(testAltAuth));
     }
 
+  @Test
+  public void cannotFindByAlternateAuthNo() {
+    //Given
+    String testAltAuth = "2222";
+    when(restTemplate.exchange(any(),any(),any(),eq(UserDto.class),eq(testAltAuth))).thenThrow(new HttpClientErrorException(HttpStatus.BAD_REQUEST));
+
+    //When
+    Optional<UserDto> result = userResourceService.findByAlternateAuthNo(testAltAuth);
+
+    //Then
+    assertFalse(result.isPresent());
+    verify(restTemplate).exchange(any(),any(),any(),eq(UserDto.class),eq(testAltAuth));
+  }
+
     @Test
-    @Ignore
     public void existsByAuthNoAndActive() {
+      //Given
+      String testAuthNo = "1111";
+      Boolean isActive = true;
+      UserDto expectedUserDto = new UserDto();
+      when(restTemplate.exchange(any(),any(),any(),eq(UserDto.class),eq(testAuthNo))).thenReturn(responseEntity);
+      when(responseEntity.getStatusCode()).thenReturn(HttpStatus.OK);
+      when(responseEntity.getBody()).thenReturn(expectedUserDto);
+
+      //When
+      Boolean result = userResourceService.existsByAuthNoAndActive(testAuthNo,isActive);
+
+      //Then
```

Comment:
this test should assert true

Reply:
It passes currently and fails if assertTrue

Reply:
fixed